File: src/pages/VolunteerLoginPage.js
import React, { useState } from "react";
import { 
  View, 
  StyleSheet, 
  Text, 
  TouchableOpacity, 
  KeyboardAvoidingView, 
  ScrollView, 
  Platform 
} from "react-native";
import { TextInput, Button } from "react-native-paper";
import AsyncStorage from "@react-native-async-storage/async-storage";

import { BASE_URL } from "../../Config";
import { getUserIdFromToken, saveLoggedInUserId } from "../utils/JWTUtils";

const DarkThemeColors = {
  background: "#111827", // Dark blue-tinted background
  primary: "#4F46E5", // Indigo for primary actions
  text: "#F3F4F6", // Off-white text
  muted: "#9CA3AF", // Muted text
  border: "#374151", // Border color
  error: "#EF4444", // Bright red for errors
  yellow: "#FBBF24", // Yellow for accents
};

const theme = {
  colors: {
    text: DarkThemeColors.text,
    placeholder: DarkThemeColors.muted,
    primary: DarkThemeColors.primary,
    background: DarkThemeColors.background,
    surface: DarkThemeColors.background,
    onSurface: DarkThemeColors.text,
    onSurfaceVariant: DarkThemeColors.text,
    surfaceVariant: DarkThemeColors.background,
  },
  roundness: 30,
};

export default function VolunteerLoginPage({ navigation }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [showPassword, setShowPassword] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const [isLoading, setIsLoading] = useState(false);

  const handleLogin = async () => {
    if (!email) {
      setErrorMessage("Email is required");
      return;
    }

    if (!password) {
      setErrorMessage("Password is required");
      return;
    }

    setErrorMessage("");
    setIsLoading(true);

    try {
      const response = await fetch(BASE_URL + "/api/login/", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          email,
          password,
        }),
      });

      const data = await response.json();

      if (!response.ok) {
        setErrorMessage(data.error || "Login failed. Please try again.");
        return;
      }

      if (data.user && data.user.account_type !== 'volunteer') {
        setErrorMessage(`Cannot continue as a volunteer. Your account type is '${data.user.account_type}'.`);
        return;
      }

      await AsyncStorage.setItem("access_token", data.access_token);
      await AsyncStorage.setItem("refresh_token", data.refresh_token);
      await AsyncStorage.setItem("user_data", JSON.stringify(data.user));

      const userId = await getUserIdFromToken();
      if (userId) {
        await saveLoggedInUserId(userId);
      } else {
        console.error("Failed to retrieve user ID from token");
      }

      navigation.navigate("VolunteerHome");
    } catch (error) {
      console.error("Login error:", error);
      setErrorMessage("Network error. Please check your connection.");
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <KeyboardAvoidingView 
      behavior={Platform.OS === "ios" ? "padding" : "height"}
      style={styles.container}
      keyboardVerticalOffset={Platform.OS === "ios" ? 64 : 0}
    >
      <ScrollView 
        contentContainerStyle={styles.scrollContainer}
        showsVerticalScrollIndicator={false}
        bounces={false}
      >
        {/* Status Bar */}
        <View style={styles.statusBar}>
          <Text style={styles.statusText}>Welcome</Text>
        </View>

        <View style={styles.logoContainer}>
          <Text style={styles.logoText}>See for Me</Text>
          <View style={styles.divider} />
          <Text style={styles.subtitle}>Vision Assistance App</Text>
          <View style={styles.labelContainer}>
            <Text style={styles.labelText}>BETA</Text>
          </View>
        </View>

        <View style={styles.formContainer}>
          <Text style={styles.formTitle}>Volunteer Login</Text>

          <TextInput
            label="Email"
            mode="outlined"
            value={email}
            onChangeText={(text) => {
              setEmail(text);
              setErrorMessage("");
            }}
            style={[styles.input, { color: DarkThemeColors.text }]}
            outlineColor={DarkThemeColors.border}
            activeOutlineColor={DarkThemeColors.primary}
            keyboardType="email-address"
            autoCapitalize="none"
            textColor={DarkThemeColors.text}
            theme={theme}
          />

          <TextInput
            label="Password"
            mode="outlined"
            value={password}
            onChangeText={(text) => {
              setPassword(text);
              setErrorMessage("");
            }}
            secureTextEntry={!showPassword}
            style={[styles.input, { color: DarkThemeColors.text }]}
            outlineColor={DarkThemeColors.border}
            activeOutlineColor={DarkThemeColors.primary}
            textColor={DarkThemeColors.text}
            theme={theme}
            right={
              <TextInput.Icon
                icon={showPassword ? "eye-off" : "eye"}
                color={DarkThemeColors.muted}
                onPress={() => setShowPassword(!showPassword)}
              />
            }
          />

          {errorMessage ? (
            <Text style={styles.errorText}>{errorMessage}</Text>
          ) : null}

          <Button
            mode="contained"
            onPress={handleLogin}
            style={styles.loginButton}
            buttonColor={DarkThemeColors.primary}
            contentStyle={styles.buttonContent}
            labelStyle={styles.buttonLabel}
            loading={isLoading}
            disabled={isLoading}
          >
            Login
          </Button>

          <View style={styles.orContainer}>
            <View style={styles.orLine} />
            <Text style={styles.orText}>OR</Text>
            <View style={styles.orLine} />
          </View>

          <Button
            mode="outlined"
            onPress={() => navigation.navigate("VolunteerSignUp")}
            style={styles.signupButton}
            labelStyle={styles.outlineButtonLabel}
            theme={{
              colors: {
                outline: DarkThemeColors.text,
              },
            }}
          >
            Create an Account
          </Button>
        </View>
      </ScrollView>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: DarkThemeColors.background,
  },
  scrollContainer: {
    flexGrow: 1,
    paddingHorizontal: 20,
    paddingBottom: 20, // Add padding at the bottom
  },
  statusBar: {
    marginTop: 40,
    alignSelf: 'center',
    alignItems: 'center',
  },
  statusText: {
    color: DarkThemeColors.muted,
    fontSize: 12,
    letterSpacing: 2,
    fontWeight: '600',
  },
  userText: {
    color: DarkThemeColors.muted,
    fontSize: 12,
    marginTop: 4,
    letterSpacing: 1,
  },
  logoContainer: {
    alignItems: "center",
    marginTop: 60, // Reduced from 100
    marginBottom: 40, // Reduced from 50
  },
  logoText: {
    fontSize: 38,
    fontWeight: "bold",
    color: DarkThemeColors.text,
    letterSpacing: 1,
  },
  divider: {
    width: 40,
    height: 2,
    backgroundColor: DarkThemeColors.border,
    marginVertical: 15,
  },
  subtitle: {
    fontSize: 16,
    color: DarkThemeColors.muted,
    marginBottom: 15,
    letterSpacing: 0.5,
  },
  labelContainer: {
    marginTop: 12,
    backgroundColor: 'rgba(251, 191, 36, 0.1)',
    paddingVertical: 4,
    paddingHorizontal: 10,
    borderRadius: 12,
  },
  labelText: {
    color: DarkThemeColors.yellow,
    fontSize: 12,
    letterSpacing: 2,
    fontWeight: '600',
  },
  formContainer: {
    width: '100%',
    paddingHorizontal: 20,
  },
  formTitle: {
    fontSize: 24,
    fontWeight: "bold",
    color: DarkThemeColors.text,
    marginBottom: 24,
    textAlign: "center",
  },
  input: {
    marginBottom: 16,
    backgroundColor: DarkThemeColors.background,
    borderRadius: 30,
  },
  loginButton: {
    borderRadius: 30,
    marginTop: 8,
    height: 50,
  },
  buttonContent: {
    height: 50,
    paddingVertical: 0,
  },
  buttonLabel: {
    color: DarkThemeColors.text,
    fontSize: 16,
    letterSpacing: 0.5,
    lineHeight: 20,
  },
  outlineButtonLabel: {
    color: DarkThemeColors.text,
    fontSize: 16,
    letterSpacing: 0.5,
  },
  orContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginVertical: 24,
  },
  orLine: {
    flex: 1,
    height: 1,
    backgroundColor: DarkThemeColors.border,
  },
  orText: {
    color: DarkThemeColors.muted,
    marginHorizontal: 16,
    fontSize: 14,
    letterSpacing: 1,
  },
  signupButton: {
    borderColor: DarkThemeColors.text,
    borderRadius: 30,
    borderWidth: 1.5,
  },
  errorText: {
    color: DarkThemeColors.error,
    fontSize: 13,
    marginBottom: 12,
    textAlign: 'center',
  },
});

File: lib/pages/volunteer_login_page.dart
// lib/pages/volunteer_login_page.dart

import 'package:flutter/material.dart';
import 'package:seeforme/pages/home_page.dart';
import 'package:seeforme/pages/volunteer_signup_page.dart';
import 'package:seeforme/services/auth_service.dart';

class VolunteerLoginPage extends StatefulWidget {
  const VolunteerLoginPage({super.key});

  @override
  State<VolunteerLoginPage> createState() => _VolunteerLoginPageState();
}

class _VolunteerLoginPageState extends State<VolunteerLoginPage> {
  final _authService = AuthService();
  final _emailController = TextEditingController();
  final _passwordController = TextEditingController();
  final _formKey = GlobalKey<FormState>();

  bool _isLoading = false;
  bool _showPassword = false;

  @override
  void dispose() {
    _emailController.dispose();
    _passwordController.dispose();
    super.dispose();
  }

  Future<void> _handleLogin() async {
    if (!_formKey.currentState!.validate()) return;

    setState(() {
      _isLoading = true;
    });

    try {
      final response = await _authService.login(
        email: _emailController.text.trim(),
        password: _passwordController.text,
      );

      // ✅ Extract tokens and userId from response
      final accessToken = response['access_token'];
      final refreshToken = response['refresh_token'];
      final userId = response['user']?['id'] ?? response['id']; // depends on API

      if (accessToken != null && refreshToken != null && userId != null) {
        await TokenStorage.saveTokensAndUser(
          accessToken: accessToken,
          refreshToken: refreshToken,
          userId: userId.toString(),
        );
      }

      Navigator.of(context).pushReplacement(
        MaterialPageRoute(builder: (context) => const VolunteerHomePage()),
      );
    } catch (e) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          backgroundColor: Colors.redAccent,
          content: Text(e.toString().replaceAll('Exception: ', '')),
        ),
      );
    } finally {
      if (mounted) {
        setState(() {
          _isLoading = false;
        });
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF111827),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.symmetric(horizontal: 20.0),
          child: Form( // Wrap in a Form for validation
            key: _formKey,
            child: Column(
              children: [
                const SizedBox(height: 40),
                const Text('Welcome', style: TextStyle(color: Color(0xFF9CA3AF), fontSize: 12, letterSpacing: 2, fontWeight: FontWeight.w600)),
                const SizedBox(height: 60),
                const Text('See for Me', style: TextStyle(fontSize: 38, fontWeight: FontWeight.bold, color: Color(0xFFF3F4F6), letterSpacing: 1)),
                Container(width: 40, height: 2, color: const Color(0xFF374151), margin: const EdgeInsets.symmetric(vertical: 15)),
                const Text('Vision Assistance App', style: TextStyle(fontSize: 16, color: Color(0xFF9CA3AF), letterSpacing: 0.5)),
                const SizedBox(height: 15),
                Container(
                  padding: const EdgeInsets.symmetric(vertical: 4, horizontal: 10),
                  decoration: BoxDecoration(color: const Color(0x1AFBBF24), borderRadius: BorderRadius.circular(12)),
                  child: const Text('BETA', style: TextStyle(color: Color(0xFFFBBF24), fontSize: 12, letterSpacing: 2, fontWeight: FontWeight.w600)),
                ),
                const SizedBox(height: 40),
                const Text('Volunteer Login', style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold, color: Color(0xFFF3F4F6))),
                const SizedBox(height: 24),
                TextFormField(
                  controller: _emailController,
                  style: const TextStyle(color: Color(0xFFF3F4F6)),
                  decoration: _inputDecoration('Email'),
                  keyboardType: TextInputType.emailAddress,
                  validator: (value) {
                    if (value == null || value.trim().isEmpty) {
                      return 'Please enter your email.';
                    }
                    return null;
                  },
                ),
                const SizedBox(height: 16),
                TextFormField(
                  controller: _passwordController,
                  style: const TextStyle(color: Color(0xFFF3F4F6)),
                  obscureText: !_showPassword,
                  decoration: _inputDecoration(
                    'Password',
                    suffixIcon: IconButton(
                      icon: Icon(_showPassword ? Icons.visibility_off : Icons.visibility, color: const Color(0xFF9CA3AF)),
                      onPressed: () {
                        setState(() {
                          _showPassword = !_showPassword;
                        });
                      },
                    ),
                  ),
                  validator: (value) {
                    if (value == null || value.isEmpty) {
                      return 'Please enter your password.';
                    }
                    return null;
                  },
                ),
                const SizedBox(height: 24),
                ElevatedButton(
                  onPressed: _isLoading ? null : _handleLogin,
                  style: ElevatedButton.styleFrom(
                    backgroundColor: const Color(0xFF4F46E5),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(30)),
                    minimumSize: const Size(double.infinity, 50),
                  ),
                  child: _isLoading
                      ? const SizedBox(height: 20, width: 20, child: CircularProgressIndicator(color: Colors.white, strokeWidth: 3))
                      : const Text('Login', style: TextStyle(color: Color(0xFFF3F4F6), fontSize: 16, letterSpacing: 0.5)),
                ),
                const SizedBox(height: 24),
                const Row(
                  children: [
                    Expanded(child: Divider(color: Color(0xFF374151))),
                    Padding(padding: EdgeInsets.symmetric(horizontal: 16.0), child: Text('OR', style: TextStyle(color: Color(0xFF9CA3AF), fontSize: 14, letterSpacing: 1))),
                    Expanded(child: Divider(color: Color(0xFF374151))),
                  ],
                ),
                const SizedBox(height: 24),
                OutlinedButton(
                  onPressed: () {
                    Navigator.of(context).push(
                      MaterialPageRoute(builder: (context) => const VolunteerSignupPage()),
                    );
                  },
                  style: OutlinedButton.styleFrom(
                    side: const BorderSide(color: Color(0xFFF3F4F6), width: 1.5),
                    shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(30)),
                    minimumSize: const Size(double.infinity, 50),
                  ),
                  child: const Text('Create an Account', style: TextStyle(color: Color(0xFFF3F4F6), fontSize: 16, letterSpacing: 0.5)),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  InputDecoration _inputDecoration(String label, {Widget? suffixIcon}) {
    return InputDecoration(
      labelText: label,
      labelStyle: const TextStyle(color: Color(0xFF9CA3AF)),
      filled: true,
      fillColor: const Color(0xFF1F2937),
      enabledBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(30), borderSide: const BorderSide(color: Color(0xFF374151))),
      focusedBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(30), borderSide: const BorderSide(color: Color(0xFF4F46E5))),
      suffixIcon: suffixIcon,
    );
  }
}